### PR TITLE
kata-deploy: validate conf file can be created

### PR DESCRIPTION
**Pull request by snir911**
As containerd doesn't exist at cleanup

Fixes: #3429
Signed-off-by: Snir Sheriber <ssheribe@redhat.com>

### CODE REVIEW

**Review by jodh-intel — APPROVED** [review]

**Comment by snir911** [comment]
The kata-deploy-ci image doesn't contain the changes made by this PR ( in kata-deploy.sh), hence, the test it keeps failing with the same error.
@fidencio @egernst @wainersm is this a known issue? can i somehow workaround just for testing?

**Comment by fidencio** [comment]
> The kata-deploy-ci image doesn't contain the changes made by this PR ( in kata-deploy.sh), hence, the test it keeps failing with the same error. @fidencio @egernst @wainersm is this a known issue? can i somehow workaround just for testing?

The way would be to manually test it, I'm afraid.

**Comment by snir911** [comment]
locally it seems to fix the issue

**Comment by wainersm** [comment]
> The kata-deploy-ci image doesn't contain the changes made by this PR ( in kata-deploy.sh), hence, the test it keeps failing with the same error. @fidencio @egernst @wainersm is this a known issue? can i somehow workaround just for testing?

In that kind of situation I do:
1. build the image and push to my quay.io
1. add a commit on top of the original PR changing the image url to the new location (create a issue to remember that you need to revert that, add a `#TODO`)
1. once the PR is merged, I re-build the image and push to the old location
1. Then I revert the patch of step 2.

Don't know this is the best optimized way possible :)

File: tools/packaging/kata-deploy/scripts/kata-deploy.sh
Comment (wainersm) [inline tools/packaging/kata-deploy/scripts/kata-deploy.sh:271]: I'm not expert on kata-deploy so caution with my suggestion: I think this whole if-then-else block needs a refactoring because it does some initialization that are specific to given actions and the actions are called just later.

For example, if `runtime` is `k3s` and action is `cleanup` then copying `containerd_conf_file` to `containerd_conf_tmpl_file` still make sense?

In other words, you fixed one case but others might be broken because the initialization is wrong there.

Comment (snir911) [reply tools/packaging/kata-deploy/scripts/kata-deploy.sh:271]: right, although in that case it's less urgent as it's not harmful.
Thing is, the difficulty of testing changes on kata-deploy makes it discouraging to touch.
I'd suggest to have currently it as is and backport to stable to unblock the release

**Review by amshinde — APPROVED** [review]

**Pull request merged**
